frontend/src/components/CreateDataSourceExample.vue

```diff
@@ -0,0 +1,202 @@
+            <template #password>
+              <span class="text-red-600">YOUR_DB_PWD</span>
+            </template>
+            <template #warehouse>
+              <span class="text-red-600">YOUR_COMPUTE_WAREHOUSE</span>
+            </template>
+          </i18n-t>
+        </template>
+        <div class="mt-2 flex flex-row">
+          <span
+            class="flex-1 min-w-0 w-full inline-flex items-center px-3 py-2 border border-r border-control-border bg-gray-50 sm:text-sm whitespace-pre"
+          >
+            {{ grantStatement(props.engineType, props.dataSourceType) }}
+          </span>
+          <button
+            tabindex="-1"
+            class="-ml-px px-2 py-2 border border-gray-300 text-sm font-medium text-control-light disabled:text-gray-300 bg-gray-50 hover:bg-gray-100 disabled:bg-gray-50 focus:ring-control focus:outline-none focus-visible:ring-2 focus:ring-offset-1 disabled:cursor-not-allowed"
+            @click.prevent="copyGrantStatement"
+          >
+            <heroicons-outline:clipboard class="w-6 h-6" />
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { defineProps, reactive, PropType } from "vue";
+import { useStore } from "vuex";
+import { toClipboard } from "@soerenmartius/vue3-clipboard";
+import { DataSourceType, EngineType } from "../types";
+import { useI18n } from "vue-i18n";
+
+interface LocalState {
+  showCreateUserExample: boolean;
+}
+
+const props = defineProps({
+  createInstanceFlag: {
+    type: Boolean,
+    default: false,
+  },
+  engineType: {
+    type: String as PropType<EngineType>,
+    default: "MYSQL",
+  },
+  dataSourceType: {
+    type: String as PropType<DataSourceType>,
+    default: "ADMIN",
+  },
+});
+
+const store = useStore();
+const { t } = useI18n();
+
+const state = reactive<LocalState>({
+  showCreateUserExample: props.createInstanceFlag,
+});
+
+const grantStatement = (
+  engineType: EngineType,
+  dataSourceType: DataSourceType
+): string => {
+  if (dataSourceType === "ADMIN") {
+    switch (engineType) {
+      case "MYSQL":
+      case "TIDB":
+        return "CREATE USER bytebase@'%' IDENTIFIED BY 'YOUR_DB_PWD';\n\nGRANT ALTER, ALTER ROUTINE, CREATE, CREATE ROUTINE, CREATE VIEW, \nDELETE, DROP, EVENT, EXECUTE, INDEX, INSERT, PROCESS, REFERENCES, \nSELECT, SHOW DATABASES, SHOW VIEW, TRIGGER, UPDATE, USAGE \nON *.* to bytebase@'%';";
+      case "CLICKHOUSE":
+        return "CREATE USER bytebase IDENTIFIED BY 'YOUR_DB_PWD';\n\nGRANT ALL ON *.* TO bytebase WITH GRANT OPTION;";
+      case "SNOWFLAKE":
+        return "CREATE OR REPLACE USER bytebase PASSWORD = 'YOUR_DB_PWD'\nDEFAULT_ROLE = \"ACCOUNTADMIN\"\nDEFAULT_WAREHOUSE = 'YOUR_COMPUTE_WAREHOUSE';\n\nGRANT ROLE \"ACCOUNTADMIN\" TO USER bytebase;";
+      case "POSTGRES":
+        return "CREATE USER bytebase WITH ENCRYPTED PASSWORD 'YOUR_DB_PWD';\n\nALTER USER bytebase WITH SUPERUSER;";
+    }
+  } else {
+    switch (engineType) {
```

Comment:
@d-bytebase @tianzhou Please check out the example code for creating a read-only user. I do some search for MySQL and ClickHouse. And don't find a short demo case for PostgreSQL or SnowFlake.

Reply:
Or we don't allow creating RO in UI for these two till we figure out the grant?

Reply:
Instead of hiding the `create` button, I prefer using a warning notification when user click it.
<img width="960" alt="image" src="https://user-images.githubusercontent.com/24653555/156906948-d3ec2196-a5c8-4870-84db-4d1ef3169e3f.png">